pennylane/ops/functions/simplify.py

```diff
@@ -62,10 +67,59 @@ def simplify(op: Operator):
     >>> qml.simplify(op)
     Adjoint(RX)(1.5707963267948966, wires=[0]) + Adjoint(PauliX)(wires=[0])
 
+    Moreover, ``qml.simplify`` can be used to simplify QNodes or quantum functions:
+
+    >>> dev = qml.device("default.qubit", wires=2)
+    >>> @qml.simplify
+        @qml.qnode(dev)
+        def circuit():
+            qml.adjoint(qml.prod(qml.RX(1, 0) ** 1, qml.RY(1, 0), qml.RZ(1, 0)))
+            return qml.probs(wires=0)
+    >>> circuit()
+    tensor([[0.64596329, 0.35403671]], requires_grad=True)
+    >>> list(circuit.tape)
+    [RZ(-1, wires=[0]) @ RY(-1, wires=[0]) @ RX(-1, wires=[0]), probs(wires=[0])]
     """
-    if qml.queuing.QueuingContext.recording():
-        with qml.tape.stop_recording():
-            new_op = copy(op.simplify())
-        qml.queuing.QueuingContext.safe_update_info(op, owner=new_op)
-        return qml.apply(new_op)
-    return op.simplify()
+    if isinstance(input, (Operator, MeasurementProcess)):
+        if QueuingContext.recording():
+            with stop_recording():
+                new_op = copy(input.simplify())
+            QueuingContext.safe_update_info(input, owner=new_op)
+            return qml.apply(new_op)
+        return input.simplify()
+
+    if isinstance(input, QuantumTape):
+        with QuantumTape() as new_tape:
+            for op in list(input):
+                _ = qml.simplify(op)
+
+        return new_tape
+
+    if callable(input):
+
+        func = input.func if isinstance(input, QNode) else input
+
+        @wraps(func)
+        def qfunc(*args, **kwargs):
+            tape = QuantumTape()
+            with stop_recording(), tape:
+                func(*args, **kwargs)
+
+            _ = [qml.simplify(op) for op in tape.operations]
+            return tuple(qml.simplify(m) for m in tape.measurements)
+
+        return (
+            QNode(
+                func=qfunc,
+                device=input.device,
+                interface=input.interface,
+                diff_method=input.diff_method,
+                expansion_strategy=input.expansion_strategy,
+                **input.execute_kwargs,
+                **input.gradient_kwargs,
+            )
+            if isinstance(input, QNode)
+            else qfunc
+        )
```

Comment:
```suggestion
        if isinstance(input, QNode):
            return (
                QNode(
                    func=qfunc,
                    device=input.device,
                    interface=input.interface,
                    diff_method=input.diff_method,
                    expansion_strategy=input.expansion_strategy,
                    **input.execute_kwargs,
                    **input.gradient_kwargs,
                )
        return qfunc
```
I think it takes too many lines to fit the `x if y else z` syntax.  But that might just be personal preference.